src/CSET/operators/ageofair.py

```diff
@@ -0,0 +1,311 @@
+                        dv = ((v * 60 * 60 * dt) / ns_spacing) * -1.0
+                        dz = (w * 60 * 60 * dt) * -1.0
+
+                    # Get column of geopot height.
+                    g_col = g_arr[(leadtime - n), :, int(y), int(x)]
+
+                    # New geopotential height of parcel - store 'capacity' between timesteps as vertical motions smaller.
+                    if n == 0:
+                        new_g = g + dz
+                        pre_g = new_g
+                    else:
+                        new_g = pre_g + dz
+
+                    # Calculate which geopot level is closest to new geopot level.
+                    z = np.argmin(np.abs(g_col - new_g))
+
+                    # Update x,y location based on displacement. Z already updated
+                    x = x + du
+                    y = y + dv
+
+                    # If it is now outside domain, then save age and dont process further with outside lam flag
+                    # Support cyclic domains like K-SCALE, where x coord out of domain gets moved through dateline.
+                    if cyclic:
+                        if (
+                            x <= -1
+                        ):  # as for example -0.3 would still be in domain, but x_arr.shape-0.3 would result in index error
+                            x = x_arr.shape[3] + x  # wrap back around dateline
+                        elif x >= x_arr.shape[3]:
+                            x = x_arr.shape[3] - x
+                    else:
+                        if x < 0 or x >= x_arr.shape[3]:
+                            ageofair_local[leadtime, lat_pnt] = n * dt
+                            outside_lam = True
+
+                    if y < 0 or y >= x_arr.shape[2]:
+                        ageofair_local[leadtime, lat_pnt] = n * dt
+                        outside_lam = True
+
+    # Save 3d array containing age of air
+    np.save(tmpdir + "/aoa_frag_" + str(lon_pnt).zfill(4) + ".npy", ageofair_local)
+
+
+def compute_ageofair(
+    cubelist, plev, includevertical=True, cyclic=False, tmpdir="/tmp/", multicore=None
+):
+    """
+    Compute back trajectories for a particular timeslice of a forecast.
+
+    Arguments:
+        cubelist: an iris.CubeList object, containing 4 cubes (u, v, w component of wind, with geopotential
+                  height. All cubes 4D, structured time, pressure, latitude, longitude. Assumed correct
+                  ordering of arrays. Expects data to already be regridded onto a standard 0p5 degree grid.
+        plev: pressure level  to evaluate the back trajectory on (etc 700 or 70000) depending on hPa or Pa.
+        includevertical: If True, use vertical velocity vector to compute back trajectories. If False, then
+                         assume flow laminar (set vertical velocity to zero).
+        tmpdir: Location of a temporary directory that can be used to write intermediate files, as used
```

Comment:
@jfrost-mo - I make use of a tmp location as part of the multiprocessing. Is there a environment variable that CSET produces for a temporary directory that I can pass to the script? I can add random hex to the start of files in case other processes are using this folder

Reply:
If this is only needed to be accessed from a single node, then use the [tempfile](https://docs.python.org/3/library/tempfile.html) library within the operator, probably [tempfile.mkdtemp](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp). That will correctly create a temporary directory in the configured system location.